x/ibc/testing/chain.go

```diff
@@ -0,0 +1,298 @@
+}
+
+// CreateClient will construct and execute a 07-tendermint MsgCreateClient. A counterparty
+// client will be created on the (target) chain.
+func (chain *TestChain) CreateClient(counterparty *TestChain) {
+	// commit counterparty's state
+	counterparty.CommitNBlocks(1)
+	chain.CommitNBlocks(1)
+
+	// construct MsgCreateClient using counterparty
+	msg := ibctmtypes.NewMsgCreateClient(
+		counterparty.ClientID, counterparty.LastHeader,
+		lite.DefaultTrustLevel, TrustingPeriod, UnbondingPeriod, MaxClockDrift,
+		chain.senderAccount.GetAddress(),
+	)
+
+	chain.SendMsg(msg)
+}
+
+// UpdateClient will construct and execute a 07-tendermint MsgUpdateClient. The counterparty
+// client will be updated on the (target) chain.
+func (chain *TestChain) UpdateClient(counterparty *TestChain) {
+	// commit counterparty's state
+	counterparty.CommitNBlocks(1)
+	chain.CommitNBlocks(1)
+
+	msg := ibctmtypes.NewMsgUpdateClient(
+		counterparty.ClientID, counterparty.LastHeader,
+		chain.senderAccount.GetAddress(),
+	)
+
+	chain.SendMsg(msg)
+}
+
+// TODO: update with msg passing
+// CreateConnection creates a connection to the counterparty with the provided state.
+func (chain *TestChain) CreateConnection(
+	connID, counterpartyConnID, clientID, counterpartyClientID string,
+	state connectiontypes.State,
+) connectiontypes.ConnectionEnd {
+
+	chain.App.BeginBlock(abci.RequestBeginBlock{Header: chain.CurrentHeader})
+
+	counterparty := connectiontypes.NewCounterparty(
+		counterpartyClientID, counterpartyConnID,
+		commitmenttypes.NewMerklePrefix(chain.App.IBCKeeper.ConnectionKeeper.GetCommitmentPrefix().Bytes()),
+	)
+
+	connection := connectiontypes.ConnectionEnd{
+		State:        state,
+		ID:           connID,
+		ClientID:     clientID,
+		Counterparty: counterparty,
+		Versions:     connectiontypes.GetCompatibleVersions(),
+	}
+
+	ctx := chain.GetContext()
+	chain.App.IBCKeeper.ConnectionKeeper.SetConnection(ctx, connID, connection)
+	return connection
+}
+
+// TODO: update with msg passing
+// CreateChannel constructs a channel with the given counterparty and connectionID.
+func (chain *TestChain) CreateChannel(
+	portID, channelID, counterpartyPortID, counterpartyChannelID string,
+	state channeltypes.State, order channeltypes.Order, connectionID string,
+) channeltypes.Channel {
+
+	counterparty := channeltypes.NewCounterparty(counterpartyPortID, counterpartyChannelID)
+
+	// sets channel with given state
+	channel := channeltypes.NewChannel(state, order, counterparty,
+		[]string{connectionID}, ChannelVersion,
+	)
+
+	ctx := chain.GetContext()
+	chain.App.IBCKeeper.ChannelKeeper.SetChannel(ctx, portID, channelID, channel)
+	chain.NewCapability(portID, channelID)
+	return channel
+}
+
+// NewCapability creates a new capability for the provided port and channel ID
+// if the capability does not already exist.
+func (chain *TestChain) NewCapability(portID, channelID string) (*capabilitytypes.Capability, error) {
```

Comment:
Why would we ever need this function directly? Shouldn't this be handled by registering ports, creating channels, etc

Reply:
yup, leftover from the old way of handling create conn/channel, will be removed once those are updated.